sdk/storage/azblob/internal/shared/shared.go

```diff
@@ -113,19 +97,35 @@ func ParseConnectionString(connectionString string) (ParsedConnectionString, err
 		suffix = defaultSuffix
 	}
 
-	if blobEndpoint, ok := connStrMap["BlobEndpoint"]; ok {
+	blobEndpoint, has_blobEndpoint := connStrMap["BlobEndpoint"]
+	accountName, has_accountName := connStrMap["AccountName"]
+
+	var serviceURL string
+	if has_blobEndpoint {
+		serviceURL = blobEndpoint
+	} else if has_accountName {
+		serviceURL = fmt.Sprintf("%v://%v.blob.%v", protocol, accountName, suffix)
+	} else {
+		return ParsedConnectionString{}, errors.New("connection string needs either AccountName or BlobEndpoint")
+	}
+
+	accountKey, has_accountKey := connStrMap["AccountKey"]
+	sharedAccessSignature, has_sharedAccessSignature := connStrMap["SharedAccessSignature"]
+
+	if has_accountName && has_accountKey {
 		return ParsedConnectionString{
-			ServiceURL:  blobEndpoint,
+			ServiceURL:  serviceURL,
 			AccountName: accountName,
 			AccountKey:  accountKey,
 		}, nil
+	} else if has_sharedAccessSignature {
+		return ParsedConnectionString{
+			ServiceURL: fmt.Sprintf("%v/?%v", serviceURL, sharedAccessSignature),
```

Comment:
Can you add a check above the return statement for checking whether the serviceURL has trailing slash. If not, then add it and update this statement.
```suggestion
			ServiceURL: fmt.Sprintf("%v?%v", serviceURL, sharedAccessSignature),
```

Reply:
Done - had to modify test data to do this though...